sheepdog/transactions/upload/sub_entities.py

```diff
@@ -383,6 +428,28 @@ def _is_valid_index_for_file(self):
             )
             is_valid = False
 
+        if not self.file_by_hash and self.file_by_uuid:
```

Comment:
this seems like a duplicated block from above. I think you can remove this and the check for `not self.use_object_id(self.entity_type)` above and it'd behave the same. It'll just populate the `record_error` with information if it can't find file by hash or uuid.

Reply:
I ran tests locally with these changes removed and everything seemed to pass... so I think it's safe to get rid of this extra logic